src/backend/access/heap/heapam.c

```diff
@@ -8724,6 +8724,7 @@ heap_xlog_visible(XLogReaderState *record)
 		PageSetAllVisible(page);
 
 		MarkBufferDirty(buffer);
+		PageSetLSN(page, lsn);
```

Comment:
Please add a comment to explain this, with a link to the pgsql-hackers thread.

Reply:
Done